Move and update appengine-endpoints-helloendpoints-java-maven sample

frankyn opened the pull request:
* This is an updated version of the [appengine-endpoints-helloendpoints-java-maven](https://github.com/GoogleCloudPlatform/appengine-endpoints-helloendpoints-java-maven) repository.
* Migrated sample to Endpoints Frameworks v2.0
* Added gradle build support
* Added support for new AppEngine plugin 
* Added comments on how to use Endpoints Frameworks v1.0. It's commented unless developer wants to use this framework version. 
* Updated README.md to reflect changes with new Endpoints Frameworks v2.0 tooling

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh:
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Wow - what a long path `endpoints-v1-using-endpoints-frameworks-v2`.  Seems quite wordy, but Ok if you don't want to use abbreviations.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/Jenkins.sh:
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Copyright 2017 Google Inc. All Rights Reserved.
```
We don't use: All Rights Reserved.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/build.gradle:
```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
```
Strike All Rights Reserved.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/UpdateApplication.launch:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Shouldn't this be Deploy? -- we may wish to drop this.

lesv replied:
Copyright  if you keep.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/eclipse-launch-profiles/DevAppServer.launch:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
We may wish to drop this, if not, it should have a copyright.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/nbactions.xml:
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
```
strike All Rights Reserved.

lesv replied:
We may wish to strike this.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<!--
+Copyright 2017 Google Inc. All Rights Reserved.
```
Strike All Rights Reserved/

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+    <packaging>war</packaging>
+    <version>1.0-SNAPSHOT</version>
+
+    <groupId>com.example.helloendpoints</groupId>
```
Is this used elsewhere?  It should be unique in the repo.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+    <artifactId>helloendpoints</artifactId>
+
+    <properties>
+        <app.id>helloendpoints</app.id>
```
YOUR-APPID-HERE

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+
+    <properties>
+        <app.id>helloendpoints</app.id>
+        <app.version>1</app.version>
```
We don't usually supply this to the new plugins.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Not needed.  Default will work.

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/pom.xml:
```diff
@@ -0,0 +1,179 @@
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
I suspect this isn't required either -- try w/o

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Constants.java:
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
```
All Rights reserved (strike)

lesv commented on appengine/endpoints-v1-using-endpoints-frameworks-v2/src/main/java/com/example/helloendpoints/Greetings.java:
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2017 Google Inc. All Rights Reserved.
```
ditto

lesv commented:
Are you ready for me to look at this yet?

frankyn commented:
@lesv PTAL

1. Acknowledging comments.
2. Adding a test to verify specific deployed version versus a prior deployment. 
3. Verified unique groupId for all repo.
4. Updated Copyrights
5. Updated README.md to reflect our conversation.

lesv commented on appengine/endpoints-frameworks-v2/README.md:
```diff
@@ -0,0 +1,19 @@
+using a discovery document development process.
+
+The new Google Cloud Endpoints Frameworks for App Engine provides
+[additional functionality][3] using OpenAPI which is a paid service. It's
```
which may require payment.

frankyn replied:
Acknowledged.

lesv commented on appengine/endpoints-frameworks-v2/discovery/Jenkins.sh:
```diff
@@ -0,0 +1,77 @@
+    "https://${1}.appspot.com/_ah/api/helloworld/v1/hellogreeting" | \
+    grep "hello world!\|goodbye world!"
+
+  # Test multiply Endpoint (This is a greeting.)
```
This doesn't look like the right format to look at a specific instance.  `name-dot-instance.appspot.com` ??

frankyn replied:
This didn't come to mind. Thanks for pointing that out!

lesv commented on appengine/endpoints-frameworks-v2/discovery/Jenkins.sh:
```diff
@@ -0,0 +1,77 @@
+function TestEndpoints () {
+  # Test getGreeting Endpoint (hello world!)
+  curl -X GET \
+    "https://${1}.appspot.com/_ah/api/helloworld/v1/hellogreeting/0" | \
```
Shouldn't this be `{$1}`

frankyn replied:
It should be ${1}.

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion

lesv commented on appengine/endpoints-frameworks-v2/discovery/Jenkins.sh:
```diff
@@ -0,0 +1,77 @@
+
+# End-2-End tests
+TestEndpoints $GOOGLE_PROJECT_ID "gradle-${GOOGLE_VERSION_ID}"
+
```
If you are going to use `${GOOGLE_VERSION_ID}` from Jenkins, you should delete your instances after you are done deploying.

frankyn replied:
Jenkins uses the GOOGLE_VERSION_ID as the test version and it expects to delete that version when tests are complete. Do you want this step to be contained within the Jenkins.sh?

lesv replied:
Yes - it hasn't been deleting things.  We ran out of versions last week.

lesv replied:
Have you run this past the google bash linter?  And made sure you are with the style guide?  https://google.github.io/styleguide/shell.xml

lesv commented on appengine/endpoints-frameworks-v2/discovery/pom.xml:
```diff
@@ -0,0 +1,124 @@
+        -->
+        <!-- End of Endpoints Frameworks v1.0 -->
+        <appengine.maven.version>1.0.0</appengine.maven.version>
+        <endpoints.framework.maven.version>1.0.0-beta2</endpoints.framework.maven.version>
```
`<endpoints.maven.plugin.version>` ??

frankyn replied:
Acknowledged.

lesv commented on appengine/endpoints-frameworks-v2/discovery/pom.xml:
```diff
@@ -0,0 +1,124 @@
+            <version>${javax.inject.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.jdo</groupId>
```
I'll need to look at your code, but I'm a bit surprised your using this.  Isn't this SQL focused?

lesv replied:
I don't see this being used anywhere, what's up???

frankyn replied:
Good catch. The sample doesn't use SQL whatsoever. I missed this dependency during clean up.

lesv commented on appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Constants.java:
```diff
@@ -0,0 +1,29 @@
+ * Contains the client IDs and scopes for allowed clients consuming the helloworld API.
+ */
+public class Constants {
+  public static final String WEB_CLIENT_ID = "replace this with your web client ID";
```
Do you mention any of this in the README??

frankyn replied:
No I did not. I updated the README with information about Google Accounts Client Ids.

lesv commented on appengine/endpoints-frameworks-v2/discovery/src/main/java/com/example/helloendpoints/Greetings.java:
```diff
@@ -0,0 +1,84 @@
+import javax.inject.Named;
+//[END begin]
+ //[START api_def]
+
```
Maybe the blank line should be between `END` & `START`  ??

frankyn replied:
Acknowledged

lesv commented on appengine/endpoints-frameworks-v2/discovery/pom.xml:
```diff
@@ -0,0 +1,124 @@
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
+    <version>1.0-SNAPSHOT</version>
+
```
Take a look at other appengine `pom.xml`, you should reference the parent `pom.xml`, then do a `mvn verify`

frankyn replied:
Acknowledged.

frankyn commented:
@lesv PTAL
